static/js/billing/helpers.js

```diff
@@ -103,27 +103,27 @@ exports.show_license_section = function (license) {
     $("#automatic_license_count").prop("disabled", true);
     $("#manual_license_count").prop("disabled", true);
 
-    const section_id = "#license-" + license + "-section";
+    const section_id = `#license-${CSS.escape(license)}-section`;
     $(section_id).show();
-    const input_id = "#" + license + "_license_count";
+    const input_id = `#${CSS.escape(license)}_license_count`;
     $(input_id).prop("disabled", false);
 };
 
 let current_page;
 
 function handle_hashchange() {
-    $("#" + current_page + '-tabs.nav a[href="' + location.hash + '"]').tab("show");
+    $(`#${CSS.escape(current_page)}-tabs.nav a[href="${CSS.escape(location.hash)}"]`).tab("show");
     $("html").scrollTop(0);
 }
 
 exports.set_tab = function (page) {
     const hash = location.hash;
     if (hash) {
-        $("#" + page + '-tabs.nav a[href="' + hash + '"]').tab("show");
+        $(`#${CSS.escape(page)}-tabs.nav a[href="${CSS.escape(hash)}"]`).tab("show");
```

Comment:
There is no need to escape "#" right?  Shouldn't this be `a[href="#${CSS.escape(hash.slice(1))}"]` instead?

Reply:
The `#` is part of the quoted string matching the `href` attribute, not part of the CSS selector syntax, so semantically it should be part of what’s being escaped. It might not matter in this case, but one should be 100% consistent and mechanical when deciding what to escape, not making case-by-case decisions that one might screw up.

Reply:
Okay. Makes sense.